### PR TITLE
Refactor unlocked_claimable_balance function to use try_fold

### PR DESCRIPTION
**Pull request by andresmechali**

### CODE REVIEW

File: frame/vesting/src/lib.rs [hunk 1]
```diff
@@ -567,20 +567,29 @@ impl<T: Config> Pallet<T> {
...
 	> {
-		let mut claims_per_schedule: BoundedBTreeMap<
-			T::VestingScheduleId,
-			BalanceOf<T>,
-			T::MaxVestingSchedules,
-		> = BoundedBTreeMap::new();
-
 		<VestingSchedules<T>>::try_mutate_exists(who, asset, |maybe_schedules| {
-			let mut total_balance_to_claim: BalanceOf<T> = Zero::zero();
 			match maybe_schedules {
```
Comment (benluelo) [inline frame/vesting/src/lib.rs:571]: this `match` could be `let schedules = maybe_schedules.ok_or(Error::<T>::VestingScheduleNotFound)?`

File: frame/vesting/src/lib.rs [hunk 2]
```diff
@@ -567,20 +567,29 @@ impl<T: Config> Pallet<T> {
...
+				Some(schedules) =>
+					vesting_schedule_ids.into_all_ids(schedules).iter().try_fold::<_, _, Result<
+						(
+							BalanceOf<T>,
+							BoundedBTreeMap<
+								T::VestingScheduleId,
+								BalanceOf<T>,
+								T::MaxVestingSchedules,
+							>,
+						),
```
Comment (benluelo) [inline frame/vesting/src/lib.rs:581]: This type annotation is reduntant - type inference should be able to pick it up (see following suggestion)

File: frame/vesting/src/lib.rs [hunk 3]
```diff
@@ -567,20 +567,29 @@ impl<T: Config> Pallet<T> {
...
+						DispatchError,
+					>>(
+						(
+							Zero::zero(),
```
Comment (benluelo) [inline frame/vesting/src/lib.rs:585]:
```suggestion
							BalanceOf::<T>::zero()
```

**Review by benluelo — COMMENTED** [review]
LGTM, just a few things :+1:

File: frame/vesting/src/lib.rs (hunk 3 above)
Comment (andresmechali) [reply frame/vesting/src/lib.rs:585]: Right, with this change the type was properly inferred.

File: frame/vesting/src/lib.rs [hunk 4]
```diff
@@ -610,16 +619,13 @@ impl<T: Config> Pallet<T> {
...
 							if locked_amount.is_zero() {
-								// remove fully claimed schedules
+								// Remove fully claimed schedules
 								schedules.remove(id_to_claim);
-							}
-
-							Ok(())
-						})
-						.collect::<Result<Vec<_>, DispatchError>>()?;
+							};
```
[the remark this answers is not in the document]
Comment (andresmechali) [reply frame/vesting/src/lib.rs:624]: I think it's fine to do it at the end, as it uses the schedule before.

**Review by PoisonPhang — APPROVED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]

**Pull request merged**